otcextensions/osclient/cbr/v3/policy.py

```diff
@@ -0,0 +1,534 @@
+
+        # optional
+        if parsed_args.day_backups:
+            attrs['operation_definition'].update(
+                day_backups=parsed_args.day_backups)
+        if parsed_args.week_backups:
+            attrs['operation_definition'].update(
+                week_backups=parsed_args.week_backups)
+        if parsed_args.month_backups:
+            attrs['operation_definition'].update(
+                month_backups=parsed_args.month_backups)
+        if parsed_args.year_backups:
+            attrs['operation_definition'].update(
+                year_backups=parsed_args.year_backups)
+        if parsed_args.max_backups:
+            attrs['operation_definition'].update(
+                max_backups=parsed_args.max_backups)
+        if parsed_args.retention_duration_days:
+            rdd = parsed_args.retention_duration_days
+            attrs['operation_definition'].update(
+                retention_duration_days=rdd)
+
+        if (parsed_args.day_backups
+            or parsed_args.week_backups
+            or parsed_args.month_backups
+                or parsed_args.year_backups) and not parsed_args.timezone:
+            msg = ("Parameter timezone must be provided if "
+                   "<day|week|month|year>_backups are being used.")
+            raise exceptions.BadRequest(msg)
+
+        if parsed_args.timezone:
+            attrs['operation_definition'].update(
+                timezone=parsed_args.timezone)
+
+        client = self.app.client_manager.cbr
+        obj = client.create_policy(**attrs)
+
+        data = utils.get_dict_properties(
+            _flatten_policy(obj), self.columns)
+
+        if obj.trigger.properties.pattern:
+            data, self.columns = _add_scheduling_patterns(
+                obj, data, self.columns)
+
+        return (self.columns, data)
+
+
+class UpdatePolicy(command.ShowOne):
+    _description = _('Update CBR Policy')
+    columns = (
+        'ID',
+        'name',
+        'operation_type',
+        'start_time',
+        'enabled',
+        'retention_duration_days',
+        'max_backups',
+        'day_backups',
+        'week_backups',
+        'month_backups',
+        'year_backups',
+        'timezone',
+    )
+
+    def get_parser(self, prog_name):
+        parser = super(UpdatePolicy, self).get_parser(prog_name)
+        parser.add_argument(
+            'policy',
+            metavar='<policy>',
+            help=_('ID or name of the CBR Policy.')
+        )
+        parser.add_argument(
+            '--name',
+            metavar='<name>',
+            help=_('Name of the CBR Policy.')
+        )
+        parser.add_argument(
+            '--disable',
+            action='store_true',
+            help=_('Disables CBR Policy which is enabled by default.')
+        )
+        parser.add_argument(
+            '--enable',
+            action='store_true',
+            help=_('Enables CBR Policy.')
+        )
```

Comment:
Should they be `store_true` both?

Reply:
Yes, your are right, but: I tried it to store the bool value in one variable (parse_args.enable) by using the 'dest' option of the parser in the '--disable' parameter. But somehow this did not work and I cannot assume why. So I used this way and dealed it differently in the action part of the code https://github.com/opentelekomcloud/python-otcextensions/blob/cbr/otcextensions/osclient/cbr/v3/policy.py#L444 .
That is not the solution I prefere, but I do not know how to solve it the way I described it above. My code always failed.